shepherd/Goal.py

```diff
@@ -67,4 +78,50 @@ def apply_powerup(self, effect, alliance):
             effect - enum representing an effect to be applied to the goal
             alliance - the alliance trying to apply the effect to the goal
         """
-        pass
+        if effect == POWERUP_TYPES.TWO_X:
+            if alliance.name == CONSTANTS.BLUE:
+                if self.blue_two_x_timer.is_running():
+                    return -1
+                self.blue_two_x_timer.start_timer(CONSTANTS.TWO_X_COOLDOWN)
+                lcm_send(CONSTANTS.LCM_TARGET_SCOREBOARD,
+                         SCOREBOARD_HEADER.POWERUPS,
+                         [self.name, alliance.name, POWERUP_TYPES.TWO_X])
+                return 1
+            elif alliance.name == CONSTANTS.GOLD:
+                if self.gold_two_x_timer.is_running():
+                    return -1
+                self.gold_two_x_timer.start_timer(CONSTANTS.TWO_X_COOLDOWN)
+                lcm_send(CONSTANTS.LCM_TARGET_SCOREBOARD,
+                         SCOREBOARD_HEADER.POWERUPS,
+                         [self.name, alliance.name, POWERUP_TYPES.TWO_X])
+                return 1
+            return -3
+        elif effect == POWERUP_TYPES.ZERO_X:
+            if alliance.name == CONSTANTS.BLUE:
+                if self.blue_zero_x_timer.is_running():
+                    return -1
+                self.blue_zero_x_timer.start_timer(
+                    CONSTANTS.ZERO_X_COOLDOWN)
+                lcm_send(CONSTANTS.LCM_TARGET_SCOREBOARD,
+                         SCOREBOARD_HEADER.POWERUPS,
+                         [self.name, alliance.name, POWERUP_TYPES.ZERO_X])
+                return 1
+            elif alliance.name == CONSTANTS.GOLD:
+                if self.gold_zero_x_timer.is_running():
+                    return -1
+                self.gold_zero_x_timer.start_timer(
+                    CONSTANTS.ZERO_X_COOLDOWN)
+                lcm_send(CONSTANTS.LCM_TARGET_SCOREBOARD,
+                         SCOREBOARD_HEADER.POWERUPS,
+                         [self.name, alliance.name, POWERUP_TYPES.ZERO_X])
+                return 1
+            return -3
```

Comment:
I think `return -3` can be generalized into just one return statement at the end.